pkg/serverless/invocationlifecycle/trace_test.go

```diff
@@ -455,3 +466,75 @@ func TestLanguageTag(t *testing.T) {
 		assert.Equal(t, tc.expectedTag, executionSpan.Meta["language"]) // expected tag from runtime
 	}
 }
+
+func TestCapturePayloadAsTags(t *testing.T) {
+	nestedMap := map[string]interface{}{
+		"key1": "value1",
+		"key2": map[string]interface{}{
+			"key3":    3,
+			"key4":    true,
+			"keylist": []interface{}{1, 2, 3, "four", 5.5, `{"keyInsideSlice":"val7","age":84}`},
+		},
+		"innerJSONString": `{"key5":"value5","age":42}`,
+		"innerJSONBytes":  []byte(`{"key6":"value6","age":21}`),
+	}
+	metaMap := make(map[string]string)
+	executionSpan := &pb.Span{
+		Meta: metaMap,
+	}
+	capturePayloadAsTags(nestedMap, executionSpan, "test", 0, 10)
+	assert.Equal(t, "value1", executionSpan.Meta["test.key1"])
+	assert.Equal(t, "3", executionSpan.Meta["test.key2.key3"])
+	assert.Equal(t, "true", executionSpan.Meta["test.key2.key4"])
+	assert.Equal(t, "1", executionSpan.Meta["test.key2.keylist.0"])
+	assert.Equal(t, "2", executionSpan.Meta["test.key2.keylist.1"])
+	assert.Equal(t, "3", executionSpan.Meta["test.key2.keylist.2"])
+	assert.Equal(t, "four", executionSpan.Meta["test.key2.keylist.3"])
+	assert.Equal(t, "5.5", executionSpan.Meta["test.key2.keylist.4"])
+	assert.Equal(t, "val7", executionSpan.Meta["test.key2.keylist.5.keyInsideSlice"])
+	assert.Equal(t, "84", executionSpan.Meta["test.key2.keylist.5.age"])
+	assert.Equal(t, "value5", executionSpan.Meta["test.innerJSONString.key5"])
+	assert.Equal(t, "42", executionSpan.Meta["test.innerJSONString.age"])
+	assert.Equal(t, "value6", executionSpan.Meta["test.innerJSONBytes.key6"])
+	assert.Equal(t, "21", executionSpan.Meta["test.innerJSONBytes.age"])
```

Comment:
It would be more complete to test the entire Meta against its expected values, which I believe the assert package can do easily. This way we're also ensuring we're not creating any extra span tags. Something like

```go
assert.Equal(t, executionSpan.Meta, map[string]string{
    "test.key1":      "value1",
    "test.key2.key3": "3",
    ... etc ...
})
```

Reply:
Nice. Updated.